mesonbuild/interpreter/mesonmain.py

```diff
@@ -235,7 +235,7 @@ def can_run_host_binaries_method(self, args: T.Tuple[object, ...], kwargs: T.Dic
         return self.can_run_host_binaries_impl(args, kwargs)
 
     def can_run_host_binaries_impl(self, args, kwargs):
-        if (self.is_cross_build_method(None, None) and
+        if (self.is_cross_build_method([], {}) and
```

Comment:
Can we just avoid calling the method? It's literally a wrapper around self.environment.is_cross_build

Reply:
You're right, added a commit that simplify that condition. `if (a) if (b) {}` is same as `if (a and b) {}`

Reply:
Also the "Need to revisit" part is outdated, it has already been revisited, that's why we have `can_run_host_binaries()` where that condition is not confusing at all.